tools/provision/provision.go

```diff
@@ -159,7 +160,7 @@ func Provision(conf Config) {
 
 		if conf.SSL {
 			var priv interface{}
-			priv, err = rsa.GenerateKey(rand.Reader, rsaBits)
+			priv, _ = rsa.GenerateKey(rand.Reader, rsaBits)
```

Comment:
It was a problem with interfaces. `error` can be converted to `sdkerror` but not the reverse.
So, if I had not defined `var err error`, then 
```
token, err := s.CreateToken(user)
```
would make err of `sdkerror`. But later, we have to use err in sending http requests, etc.
That's why to avoid creating new variable of `sdkerr`, I defined err var at the top.

As for removing the `err` to `_`

```
var priv interface{}
			priv, err = rsa.GenerateKey(rand.Reader, rsaBits)
			priv, _ = rsa.GenerateKey(rand.Reader, rsaBits)

			notBefore := time.Now()
			validFor, err := time.ParseDuration(ttl)
```
We weren't actually using that value of err, so it was showing a 'problem' as that value of err was never being used. So I removed it for the time being.

Reply:
Since `sdkerror` implements  `error`, so it's compatible, but since it's not handled anyway, it's a separate issue in itself and should not be a part of this PR. I'll open a new issue and resolve the comment.